async-http-client/src/main/scala/org/http4s/client/asynchttpclient/AsyncHttpClient.scala

```diff
@@ -88,8 +88,9 @@ object AsyncHttpClient {
           subscriber <- StreamSubscriber[F, HttpResponseBodyPart]
 
           subscribeF = F.delay(publisher.subscribe(subscriber))
+
           bodyDisposal <- Ref.of[F, F[Unit]] {
-            subscriber.stream(subscribeF).take(0).compile.drain
+            subscriber.stream(subscribeF).pull.uncons.void.stream.compile.drain
```

Comment:
Can you please explain what "force finalization" means, @wemrysi?

Reply:
@kevinmeredith in our usage, we noticed that if the body of responses produced by the AHC client weren't consumed, the connection was never closed. A solution to this was to add an effect to the `Resource` finalizer that would subscribe to the body stream and then immediately `cancel` the subscription, which closes the connection.

The original attempt at this used `take(0)` which turns out to be a noop (it results in a pure, empty stream) and thus the subscribe/cancel effects are never run. Switching to `pull.uncons` ensures that the first effect in the stream is sequenced and any finalizers are run when the stream ends.

Reply:
Thanks @wemrysi!